### ISSUE TITLE
TODO: wrappers for level, fs, and net, setTimeout etc

### ISSUE THREAD

**Issue by dominictarr** [dominictarr/interleavings#1]
it would be greate to create in memory mocks of various async node apis,
because this would make node.js deterministic, and thus we'd be able to run pretty much any test through interleavings and search for race conditions.

this could be particularily useful for things like ssb @pfraze
I'd expect that the level wrapper would be the easiest, since there are already memory only leveldowns that could be adapted.

it would probably be easiest to write a mock for the net binding:
https://github.com/nodejs/io.js/blob/master/lib/net.js#L14-L19
only a few things to replace.

fs would need a few things:
`binding.{stat, access, open, read, close, fstat, lstat, writeBuffer, writeString, rename, unlink, ftruncate, readdir, mkdir, rmdir, fdatasync, fsync, readlink, symlink, link, fchmod, chmod, fchown, chown, utimes, futimes, StatWatcher}` okay so that is actually quite a lot.

**Comment by dominictarr** [comment]
yeah, so there would be quite a bit of work here, but it would be pretty straightforward, since there is well defined api and behaviour so there isn't any design work.

**Comment by pfraze** [comment]
:+1: very good idea
